src/TrackGenerator.cpp

```diff
@@ -1918,14 +2024,29 @@ void TrackGenerator::initializeSegments() {
   FSRs_to_keys = _geometry->getFSRsToKeys();
 
   /* Iterate over all Track segments and assign them each a Material */
-  for (int i=0; i < _num_azim; i++) {
-    for (int j=0; j < _num_tracks[i]; j++) {
-      for (int s=0; s < _tracks[i][j].getNumSegments(); s++) {
-        curr_segment = _tracks[i][j].getSegment(s);
-	region_id = curr_segment->_region_id;
-        Material* mat = _geometry->findFSRMaterial(region_id);
-	curr_segment->_material = _geometry->findFSRMaterial(region_id);
-	FSR_keys_map->at(FSRs_to_keys->at(region_id))->_mat_id = mat->getId();
+#pragma omp parallel
```

Comment:
I like the performance improvements to this routine a lot. As above, I'm wary of this simply because I added this functionality in to support parametric studies of multiple simulations with different MGXS libraries. We probably need a few "multi-sim" tests w/ and w/o CMFD with different MGXS to demonstrate that initialization and memory deallocation still work properly with parallelization for these scenarios as they do now. I don't think the onus needs to be on you to put in these tests for this PR, but perhaps we should hold off on merging this until I can patch together 3-5 "multi-sim" tests in a separate PR.